### DISCUSSION TITLE
The ..currentTiddler and currentTiddler inside a Function

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#8227] [General]
* open https://tiddlywiki.com/
* Create a tiddler with below script
```xml
\procedure getType() [<..currentTiddler>addsuffix[ → ]addsuffix<currentTiddler>]
\function f.getType() [<..currentTiddler>addsuffix[ → ]addsuffix<currentTiddler>]

--- Direct filter

<$tiddler tiddler="HelloThere">
<$list
  filter="[all[current]links[]limit[5]]  
          :map[<..currentTiddler>addsuffix[ → ]addsuffix<currentTiddler>]">
<<currentTiddler>><br>
</$list>

</$tiddler>

--- With procedure

<$tiddler tiddler="HelloThere">
<$list
  filter="[all[current]links[]limit[5]]  
          :map[subfilter<getType>]">
<<currentTiddler>><br>
</$list>

</$tiddler>

--- With function

<$tiddler tiddler="HelloThere">
<$list
  filter="[all[current]links[]limit[5]]  
          :map[f.getType[]]">
<<currentTiddler>><br>
</$list>

</$tiddler>
```

This will produces

```
— Direct filter

HelloThere → TiddlyWiki
HelloThere → Philosophy of Tiddlers
HelloThere → Creating and editing tiddlers
HelloThere → Structuring TiddlyWiki
HelloThere → Sharing your tiddlers with others
— With procedure

HelloThere → TiddlyWiki
HelloThere → Philosophy of Tiddlers
HelloThere → Creating and editing tiddlers
HelloThere → Structuring TiddlyWiki
HelloThere → Sharing your tiddlers with others
— With function

→ HelloThere
→ HelloThere
→ HelloThere
→ HelloThere
→ HelloThere
```

That means the `..currentTiddler` is not known inside a function and `currentTiddler` also refers to the value of the variable currentTiddler outside the filter run. Note that in the first two examples (direct use of ..currentTiddler in :map and subfilter<procname>) work as expected.

**NOTE**
If I change function like this: `\function f.getType() [get[title]addprefix[ → ]addprefix<currentTiddler>]`, it will produce the correct answer.

**Question**
Is this the correct behaviour for function used inside a filter run prefix (here :map, but :sort and other filter run prefix may be used)?

**Comment by saqimtiaz** [comment]
There is a know bug to do with functions and the value of currentTiddler, see https://github.com/Jermolene/TiddlyWiki5/issues/8178

**Reply by kookma** [reply]
Thank you @saqimtiaz 
Testing on https://tiddlywiki.com/prerelease/ seems my above function works correctly!
